lib/components/narrative/narrative-itineraries.js

```diff
@@ -30,48 +33,174 @@ class NarrativeItineraries extends Component {
     itineraryClass: DefaultItinerary
   }
 
+  state = {}
+
+  _toggleDetailedItinerary = () => {
+    this.setState({showDetails: !this.state.showDetails})
+  }
+
+  _saveTrip = () => {
+    window.confirm('Are you sure you want to save this trip?')
+  }
+
+  _onFilterChange = evt => {
+    const {sort, updateItineraryFilter} = this.props
+    const {value} = evt.target
+    updateItineraryFilter({filter: value, sort})
+  }
+
+  _onSortChange = evt => {
+    const {value: type} = evt.target
+    const {filter, sort, updateItineraryFilter} = this.props
+    updateItineraryFilter({filter, sort: {...sort, type}})
+  }
+
+  _onSortDirChange = () => {
+    const {filter, sort, updateItineraryFilter} = this.props
+    const direction = sort.direction === 'ASC' ? 'DESC' : 'ASC'
+    updateItineraryFilter({filter, sort: {...sort, direction}})
+  }
+
   _toggleRealtimeItineraryClick = (e) => {
     const {setUseRealtimeResponse, useRealtime} = this.props
     setUseRealtimeResponse({useRealtime: !useRealtime})
   }
 
+  _renderLoadingDivs = () => {
+    const {itineraries, modes, pending} = this.props
+    if (!pending) return null
+    // Construct loading divs as placeholders while all itineraries load.
+    const count = modes.combinations
+      ? modes.combinations.length - itineraries.length
+      : 0
+    return Array.from(
+      {length: count},
+      (v, i) =>
+        <div key={i} className='option default-itin'>
+          <SkeletonTheme color='#ddd' highlightColor='#eee'>
+            <Skeleton count={3} />
+          </SkeletonTheme>
+        </div>
+    )
+  }
+
   render () {
     const {
       activeItinerary,
+      activeSearch,
+      containerStyle,
+      filter,
       itineraries,
       itineraryClass,
+      pending,
+      persistence,
       realtimeEffects,
+      sort,
       useRealtime
     } = this.props
-    if (!itineraries) return null
-
+    if (!activeSearch) return null
+    const itineraryIsExpanded = activeItinerary !== undefined && activeItinerary !== null && this.state.showDetails
+    const showRealtimeAnnotation = realtimeEffects.isAffectedByRealtimeData && (
+      realtimeEffects.exceedsThreshold ||
+      realtimeEffects.routesDiffer ||
+      !useRealtime
+    )
+    const resultText = pending
+      ? 'Finding your options...'
+      : `${itineraries.length} itineraries found.`
```

Comment:
Doesn't ever show an error message if one is returned in OTP response.

See usage in narrative-routing-results

https://github.com/opentripplanner/otp-react-redux/blob/a6b1710190b15499bd8b8fde9fc2d2916d0f8b52/lib/components/narrative/narrative-routing-results.js#L30

Reply:
fixed.